lib/valueflow.cpp

```diff
@@ -3044,254 +3044,283 @@ static void valueFlowAfterAssign(TokenList *tokenlist, SymbolDatabase* symboldat
+
+                    // After conditional code..
+                    if (Token::simpleMatch(top->link(), ") {")) {
+                        Token *after = top->link()->linkAt(1);
+                        std::string unknownFunction;
+                        if (settings->library.isScopeNoReturn(after, &unknownFunction)) {
+                            if (settings->debugwarnings && !unknownFunction.empty())
                                 bailout(tokenlist, errorLogger, after, "possible noreturn scope");
                             continue;
                         }
-                        dead_else = isReturnScope(after);
-                    }
 
-                    std::list<ValueFlow::Value> * values = nullptr;
-                    if (!dead_if && check_if)
-                        values = &true_values;
-                    else if (!dead_else && check_else)
-                        values = &false_values;
-
-                    if (values) {
-                        // TODO: constValue could be true if there are no assignments in the conditional blocks and
-                        //       perhaps if there are no && and no || in the condition
-                        bool constValue = false;
-                        valueFlowForward(after->next(), top->scope()->bodyEnd, var, varid, *values, constValue, false, tokenlist, errorLogger, settings);
+                        const bool dead_if = isReturnScope(after);
+                        bool dead_else = false;
+
+                        if (Token::simpleMatch(after, "} else {")) {
+                            after = after->linkAt(2);
+                            if (Token::simpleMatch(after->tokAt(-2), ") ; }")) {
+                                if (settings->debugwarnings)
+                                    bailout(tokenlist, errorLogger, after, "possible noreturn scope");
+                                continue;
+                            }
+                            dead_else = isReturnScope(after);
+                        }
+
+                        std::list<ValueFlow::Value> *values = nullptr;
+                        if (!dead_if && check_if)
+                            values = &cond.true_values;
+                        else if (!dead_else && check_else)
+                            values = &cond.false_values;
+
+                        if (values) {
+                            // TODO: constValue could be true if there are no assignments in the conditional blocks and
+                            //       perhaps if there are no && and no || in the condition
+                            bool constValue = false;
+                            forward(after, top->scope()->bodyEnd, var, *values, constValue);
+                        }
                     }
                 }
             }
         }
     }
+};
+
+static void setConditionalValues(const Token *tok,
+                                 bool invert,
+                                 long long value,
```

Comment:
You should use `Mathlib::bigint` instead of `long long`. We might someday change it to `__int128` or some other integer type with more bits.